src/os/bluestore/NVMEDevice.cc

```diff
@@ -986,10 +986,40 @@ void NVMEDevice::aio_submit(IOContext *ioc)
     ioc->num_pending -= pending;
     assert(ioc->num_pending.load() == 0);  // we should be only thread doing this
     // Only need to push the first entry
-  if(queue_id == -1)
-    queue_id = ceph_gettid();
-    driver->get_queue(queue_id)->queue_task(t, pending);
+    if(queue_id == -1)
+      queue_id = ceph_gettid();
     ioc->nvme_task_first = ioc->nvme_task_last = nullptr;
+    driver->get_queue(queue_id)->queue_task(t, pending);
```

Comment:
could you let us know why you switch the order of the two lines above?

Reply:
clear the ioc first will be safe since we will not use these pointer any more, since t is already assigned value from nvme_task_first. Before submitting the I/O to be handled in the spdk thread, we  should better set it to NULL